src/main/python/ayab/scene.py

```diff
@@ -152,45 +157,45 @@ def refresh(self):
         self.setScene(qscene)
 
     @property
-    def row_progress(self):
+    def row_progress(self)->int:
         return self.__row_progress
 
     @row_progress.setter
-    def row_progress(self, row_progress):
+    def row_progress(self, row_progress:int)->None:
         self.__row_progress = row_progress
         self.refresh()
 
-    def update_needles(self, start_needle, stop_needle):
+    def update_needles(self, start_needle:int, stop_needle:int)->None:
         '''Update the position of the start/stop needle visualization'''
         self.__start_needle = start_needle
         self.__stop_needle = stop_needle
         self.refresh()
 
     @property
-    def alignment(self):
+    def our_alignment(self)->Alignment:
         return self.__alignment
 
-    def update_alignment(self, alignment):
+    def update_alignment(self, alignment:Alignment)->None:
         '''Update the alignment of the image between start/stop needle'''
         self.__alignment = Alignment(alignment)
         self.refresh()
 
-    def wheelEvent(self, event):
+    def wheelEvent(self, event:QWheelEvent)->None:
         '''Zoom the pattern upon mouse wheel event'''
-        self.zoom = event
+        self.zoom = event #type: ignore
```

Comment:
mypy does not understand complex setters so we are once again forced to ignore typing here.